app/assets/javascripts/static.js

```diff
@@ -37,93 +28,292 @@ $( document ).ready(function() {
         $("div.search div.start-year span.hint").html("Start Year")
         $("div.search input#dash").removeClass("hidden");
         $("div.search div.end-year").removeClass("hidden");
         $("div.search div.end-month").removeClass("hidden");
         $("div.search div.end-date").removeClass("hidden");
-        $('button.range-toggle span.range-toggle-text').html("Show Specific Date");
-        $('button.range-toggle').val("Show Single Date");
-        $('button.range-toggle  span.glyphicon-chevron-left').removeClass("hidden");
-        $('button.range-toggle  span.glyphicon-chevron-right').addClass("hidden");
+        $('div.search button.range-toggle span.range-toggle-text').html("Show Specific Date");
+        $('div.search button.range-toggle').val("Show Single Date");
+        $('div.search button.range-toggle  span.glyphicon-chevron-left').removeClass("hidden");
+        $('div.search button.range-toggle  span.glyphicon-chevron-right').addClass("hidden");
     } else {
         $("div.search div.start-year span.hint").html("Specific date")
         $("div.search input#dash").addClass("hidden");
         $("div.search div.end-year").addClass("hidden");
         $("div.search div.end-month").addClass("hidden");
         $("div.search div.end-date").addClass("hidden");
-        $('button.range-toggle span.range-toggle-text').html("Show Date Range");
-        $('button.range-toggle').val("Show Date Range");
-        $('button.range-toggle  span.glyphicon-chevron-right').removeClass("hidden");
-        $('button.range-toggle  span.glyphicon-chevron-left').addClass("hidden");
+        $('div.search button.range-toggle span.range-toggle-text').html("Show Date Range");
+        $('div.search button.range-toggle').val("Show Date Range");
+        $('div.search button.range-toggle  span.glyphicon-chevron-right').removeClass("hidden");
+        $('div.search button.range-toggle  span.glyphicon-chevron-left').addClass("hidden");
     }
     
   });  
 
-  $("div.search form input, div.search form select").on("change", function(){
+  $("div.search form input.changeable, div.search form select.changeable").on("change", function(event){
+
+    // alert("id: " + this.id);
+    // alert("value: " + this.value);
+
+    if ( this.id  == "range_start_month" ) {
+       update_month_dates("div.search form input#range_start_year", "div.search form input#range_start", "div.search form select#range_start_month", "div.search form select#range_start_date");
+    }
+
+    if (this.id == "range_end_month") {
+        update_month_dates("div.search form input#range_end_year", "div.search form input#range_end", "div.search form select#range_end_month", "div.search form select#range_end_date");
+    }
+
+    set_dates("div.search ");
+    event.preventDefault();
+
+  });
+
+  $("div.search form input, div.search form select").on("load", function(event){
     
-    set_dates();
+    set_dates("div.search ");
     event.preventDefault();
```

Comment:
Do these elements get a `load` event? I'm not sure you need this function handler.